src/components/CustomRpcSettingsForm.vue

```diff
@@ -58,35 +49,69 @@ export default {
     }
   },
   computed: {
+    ...mapState(['activeNetwork', 'activeWalletId']),
     canSubmit() {
       const { networkName, newRpcUrl, chainId, currencySymbol } = this.formData
       return !!(networkName && newRpcUrl && chainId && currencySymbol)
     }
   },
   methods: {
-    formSubmitHandler() {
-      // TODO :: Handle functionality from vuex (inside wallet-core)
-      console.log('Save changes', this.formData)
+    ...mapActions(['saveCustomChainSettings', 'removeCustomChainSettings']),
+    async formSubmitHandler() {
+      const { chain, network, asset } = this.settings
+      const { name,
+        coinType,
+        isTestnet,
+        chainId,
+        rpcUrl,
+        helperUrl,
+        custom } = network
+
+      const payload = {
+        network: this.activeNetwork,
+        walletId: this.activeWalletId,
+        chainId,
+        chanifyNetwork: {
+          ...network,
```

Comment:
@bradleySuira If we are pulling all data from the network object here, we will need to remove lines 62 to 68